code/modules/events/apc_short.dm

```diff
@@ -26,55 +26,108 @@
 /datum/event/apc_short/end()
 	return TRUE
 
-/proc/power_failure(announce=TRUE)
+/proc/depower_apcs()
 	var/list/skipped_areas_apc = list(
 		/area/engine/engineering,
 		/area/turret_protected/ai)
+	GLOB.event_announcement.Announce("Power failure detected in [station_name()]'s powernet. All APCs have lost power. Gravity systems likely to fail.", "Systems Power Failure", new_sound = 'sound/AI/attention.ogg')
+	for(var/thing in GLOB.apcs)
+		var/obj/machinery/power/apc/A = thing
+		var/area/current_area = get_area(A)
+		if((current_area.type in skipped_areas_apc) || !is_station_level(A.z))
+			continue
+		var/obj/item/stock_parts/cell/C = A.get_cell()
+		if(C)
+			C.charge = 0
+		current_area.power_change()
+	log_and_message_admins("Power has been drained from all APCs.")
 
+/proc/power_failure(announce=TRUE)
+	// skip any APCs that are too critical to disable
+	var/list/skipped_areas_apc = list(
+		/area/engine/engineering,
+		/area/turret_protected/ai)
 	if(announce)
 		GLOB.event_announcement.Announce("Overload detected in [station_name()]'s powernet. Engineering, please repair shorted APCs.", "Systems Power Failure", new_sound = 'sound/AI/attention.ogg')
-
 	// break APC_BREAK_PROBABILITY% of all of the APCs on the station
 	var/affected_apc_count = 0
-	for(var/thing in GLOB.apcs)
-		var/obj/machinery/power/apc/C = thing
-		// skip any APCs that are too critical to disable
-		var/area/current_area = get_area(C)
-		if((current_area.type in skipped_areas_apc) || !is_station_level(C.z))
+	for(var/obj/machinery/power/apc/A in GLOB.apcs)
+		var/area/current_area = get_area(A)
+		if((current_area.type in skipped_areas_apc) || !is_station_level(A.z))
 			continue
 		// if we are going to break this one
 		if(prob(APC_BREAK_PROBABILITY))
 			// if it has internal wires, cut the power wires
-			if(C.wires)
-				if(!C.wires.is_cut(WIRE_MAIN_POWER1))
-					C.wires.cut(WIRE_MAIN_POWER1)
-				if(!C.wires.is_cut(WIRE_MAIN_POWER2))
-					C.wires.cut(WIRE_MAIN_POWER2)
-			// if it was operating, toggle off the breaker
-			if(C.operating)
-				C.toggle_breaker()
+			if(A.wires)
+				if(!A.wires.is_cut(WIRE_MAIN_POWER1))
+					A.wires.cut(WIRE_MAIN_POWER1)
+				if(!A.wires.is_cut(WIRE_MAIN_POWER2))
+					A.wires.cut(WIRE_MAIN_POWER2)
+				// if it was operating, toggle off the breaker
+				if(A.operating)
+					A.toggle_breaker()
```

Comment:
Why did line 67 and 68 get indented?

Reply:
That's curious. According to Gitlens, it look I cute a lot of the code out and then pasted it back in. If I recall correctly, I was trying to adjust it so admins had options both for depowering the station _and_ powering the station. In the midst of rearranging the code, I must have missed that indent. I should have caught that and did not.